### PR TITLE
Refactor: address remaining issues found by Spearbit audit

### PR DESCRIPTION
**Pull request by amusingaxl**
- docs(StakingRewards): add warning regarding `notifyRewardAmount` usage
- refactor(StakingRewards): add check in constructor

### CODE REVIEW

**Comment by oldchili** [comment]
Should this PR also revert https://github.com/makerdao/endgame-toolkit/commit/8ff5f3935eab27ebbadf54cbe9057facd5704f3c ?

**Comment by amusingaxl** [comment]
> Should this PR also revert [8ff5f39](https://github.com/makerdao/endgame-toolkit/commit/8ff5f3935eab27ebbadf54cbe9057facd5704f3c) ?

I don't believe so. We usually want to avoid failing scripts. At the very least, it prevents the deployer from wasting some gas in case of a problem in the config.
The script is also some kind of documentation, even though in this case it is pretty obvious from the code, but that is not always the case.

File: README.md [hunk 1]
```diff
@@ -69,6 +69,14 @@ through a [`StakingRewards`](#stakingrewards) contract.
...
   - This allows the duration to be changed even when new reward is added immediately after the end of a distribution
     period, which would be the case when `rewardDistribution` is a smart contract distributing new reward every period.
+- Prevent `rewardsToken` and `stakingToken` from being the same token.
+  - Instead of using the _pull pattern_, this contract expect the address with the `rewardsDistribution` role to
+    transfer the tokens to it and call `notifyRewardAmount` with the correct value.
+  - There is a possible scenario where the amount of tokens transferred to the contract is less than the amount used as
+    a parameter in `notifyRewardAmount`. In such case, if the staking token and rewards token are the same, the staked
+    tokens would be incorporated into the new reward period, causing stakers to lose their capital.
+
+Full diff is referenced [here](./src/synthetix/StakingRewards.sol#L8).
```
Comment (oldchili) [inline README.md]: Up to you, but I'd just remove the diff links from everywhere and would just leave the permalink to the original code (in one place). Anyone can do the diff given the permalink. 
IMO the code should be easily maintainable, which means not having to look for many places to update the documentation on each change.

File: src/synthetix/StakingRewards.sol [hunk 2]
```diff
@@ -135,6 +137,11 @@ contract StakingRewards is IStakingRewards, Pausable, ReentrancyGuard {
 
     /* ========== RESTRICTED FUNCTIONS ========== */
 
+    /// @dev Before calling this function, the caller should send at least `reward` tokens to the contract.
```
Comment (oldchili) [inline src/synthetix/StakingRewards.sol]: Otherwhise => Otherwise

Comment (oldchili) [inline src/synthetix/StakingRewards.sol]: Neat - I wouldn't put the @dev natspec tag, as natspec is not used in other places in this contract.

**Review by sunbreak1211 — COMMENTED** [review]
PR looks good to me, I'll wait for the typo fix and response in the other comment to give my approval.

**Comment by telome** [comment]
LGTM, will approve once the comments above are addressed.

File: README.md (hunk 1 above)
Comment (amusingaxl) [reply README.md]: Done.

File: src/synthetix/StakingRewards.sol (hunk 2 above)
Comment (amusingaxl) [reply src/synthetix/StakingRewards.sol]: Fixed.

Comment (amusingaxl) [reply src/synthetix/StakingRewards.sol]: Updated.

File: README.md (hunk 1 above)
Comment (oldchili) [reply README.md]: I meant removing also from StakingRewards.sol.

**Review by oldchili — APPROVED** [review]
LGTM

**Review by sunbreak1211 — APPROVED** [review]
LGTM

**Review by telome — APPROVED** [review]
LGTM

**Pull request merged**